internal/maintenance/queue_cleaner.go

```diff
@@ -0,0 +1,162 @@
+package maintenance
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"strings"
+	"time"
+
+	"github.com/riverqueue/river/internal/baseservice"
+	"github.com/riverqueue/river/internal/maintenance/startstop"
+	"github.com/riverqueue/river/internal/rivercommon"
+	"github.com/riverqueue/river/internal/util/timeutil"
+	"github.com/riverqueue/river/internal/util/valutil"
+	"github.com/riverqueue/river/riverdriver"
+)
+
+const (
+	QueueCleanerIntervalDefault = time.Hour
+	QueueRetentionPeriodDefault = 24 * time.Hour
```

Comment:
Add "cleaner" to the name of this one, and for the time being, maybe unexport both constants. Neither are needed outside the packages and IMO no reason to ever really make these configurable unless there's a really amazing reason to do so.

Reply:
`QueueCleanerRetentionPeriodDefault` (née `QueueRetentionPeriodDefault`) is actually referenced from related client tests in two places, and in the `Client` maintenance initialization. Would you prefer that this config not be exposed and that the `QueueCleaner` apply its own default without exposing that value to the main package?